### PR TITLE
DolphinQt: Add Show/Hide All options to gamelist region menu

### PR DESCRIPTION
**Pull request by Dentomologist**
If you wish to only display games from only one region you currently have to open the View -> Regions menu 13 times to manually deselect every other region.  This PR adds "Show All" and "Hide all" options to simplify the process.

image

### CODE REVIEW

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DolphinQt/MenuBar.cpp [hunk 1]
```diff
@@ -695,20 +695,29 @@ void MenuBar::AddShowRegionsMenu(QMenu* view_menu)
...
-    connect(action, &QAction::toggled, [this, config, key](bool value) {
-      *config = value;
-      emit GameListRegionVisibilityToggled(key, value);
-    });
+    bool* const config = region_map[key];
+    QAction* const menu_item = region_menu->addAction(key);
+    menu_item->setCheckable(true);
+    menu_item->setChecked(*config);
+
+    const auto SetVisibility = [this, config, key, menu_item](bool visibility) {
```
Comment (leoetlino) [inline Source/Core/DolphinQt/MenuBar.cpp:710]:
```suggestion
    const auto set_visibility = [this, config, key, menu_item](bool visibility) {
```

> [2020-10-29 16:44:05] Lioncash: I generally write the lambda name as `snake_case` like a variable, as opposed to `CamelCase` like we do for functions, so it's visually obvious that a lambda is likely being invoked. (and also doesn't add another special case to variable naming to remember).

File: Source/Core/DolphinQt/MenuBar.cpp [hunk 2]
```diff
@@ -695,20 +695,29 @@ void MenuBar::AddShowRegionsMenu(QMenu* view_menu)
...
+      menu_item->setChecked(visibility);
+      *config = visibility;
+      emit GameListRegionVisibilityToggled(key, visibility);
+    };
+    const auto SetVisible = std::bind(SetVisibility, true);
```
Comment (leoetlino) [inline Source/Core/DolphinQt/MenuBar.cpp:715]:
```suggestion
    const auto set_visible = std::bind(SetVisibility, true);
```

File: Source/Core/DolphinQt/MenuBar.cpp [hunk 3]
```diff
@@ -695,20 +695,29 @@ void MenuBar::AddShowRegionsMenu(QMenu* view_menu)
...
+    const auto SetHidden = std::bind(SetVisibility, false);
```
Comment (leoetlino) [inline Source/Core/DolphinQt/MenuBar.cpp:716]:
```suggestion
    const auto set_hidden = std::bind(SetVisibility, false);
```

**Review by leoetlino — CHANGES_REQUESTED** [review]

File: Source/Core/DolphinQt/MenuBar.cpp (hunk 2 above)
Comment (Dentomologist) [reply Source/Core/DolphinQt/MenuBar.cpp]: I needed to change the SetVisibility on this line and set_hidden, too.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**